packages/primitives-core/src/styled.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+import * as React from 'react'
+import {
+  channel,
+  contextTypes,
+  setTheme,
+  testAlwaysTrue,
+  pickAssign,
+  interleave
+} from './utils'
+import { createCss } from './css'
+
+type State = {
+  theme: Object
+} | null
+
+let defaultPickTest = prop => prop !== 'theme' && prop !== 'innerRef'
+
+type options = {
+  getShouldForwardProp: (cmp: React.ElementType) => (prop: string) => boolean
+}
+
+export function createStyled(
+  StyleSheet: Object,
+  { getShouldForwardProp = () => defaultPickTest }: options = {}
```

Comment:
I've noticed that this API differs from web one. Regular emotion accepts `shouldForwardProp` - is there any reason why this is different here?

Reply:
the whole `createStyled` also differs from regular `create-emotion-styled` API which accepts whole emotion (created upfront with `create-emotion`). It seems to me that react-native equivalent would be
```js
const css = createCss(StyleSheet)
const styled = createStyled(css)
```
or 
```js
const css = createCss(StyleSheet)
const styled = createStyled({ css })
```
I'm not sure if native version has some other API than `css`/`styled` (web one has), so I'm not sure which one would be more appropriate here.

Reply:
regarding `shouldForwardProp` - I think I've found the answer, this is used at 1 layer above than regular `shouldForwardProp`, right? would be good to support normal `shouldForwardProp`, but it can also be added later (maybe create some kind of TODO list of things that could be added to achieve parity with regular `emotion`)

Reply:
> The API here differs because, we may add some more options along with getShouldForwardProp in the future.

Making it an object is ofc preferrable, but that was not the question. Normal `emotion` also accepts configuration objects - my question was about API differences.

And what about 1st arg to `createStyled`?

Reply:
There are no such API differences except we use `StyleSheet` constructor and a little refactored code taken from `@emotion/primitives`. The goal is to keep the signature of functions similar across the env. whether it's web, native or sketch. I liked your suggestion and will make the changes.

Reply:
Your argument is valid here since having such API reduces the maintenance cost.